scripts/generate_pinval/generate_pinval.py

```diff
@@ -499,6 +561,54 @@ def _fmt(x):
     return formatted_df
 
 
+def compute_shap_weights(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
This function is the meat of the PR! I wish we could do this in SQL so we could test it with dbt tests, but I think the speed that pandas offers us is more important than the reliability of SQL right now.